test/models/test_video_vqvae.py

```diff
@@ -47,3 +67,105 @@ def test_forward(self, random):
             ]
         )
         assert_expected(actual, expected, rtol=0, atol=1e-4)
+
+
+class TestVideoEncoder:
+    @pytest.fixture(scope="class")
+    def encoder(self, random, in_channels, out_channels):
+        return VideoEncoder(
+            in_channels=in_channels[:2],
+            out_channels=out_channels[:2],
+            kernel_sizes=[2, 2],
+            strides=[1, 1],
+            n_res_layers=1,
+            embedding_dim=2,
+        )
+
+    def test_invalid_arg_length(self, in_channels, out_channels):
+        with pytest.raises(ValueError):
+            _ = VideoEncoder(in_channels[:2], out_channels, [2, 2], [1, 1], 1, 1)
+
+    def test_invalid_in_out_channels(self, in_channels, out_channels):
+        with pytest.raises(ValueError):
+            _ = VideoEncoder(in_channels, out_channels, [2, 2, 2], [1, 1, 1], 1, 1)
+
+    def test_forward(self, input_tensor, encoder):
+        actual = encoder(input_tensor)
+        expected = torch.tensor(
+            [
+                [
+                    [
+                        [[0.7367, 0.3994], [0.3994, 0.3890]],
+                        [[0.5026, 0.1018], [1.1798, 1.2604]],
+                    ],
+                    [
+                        [[0.4504, 0.2898], [0.2898, 0.2811]],
+                        [[0.3761, 0.0409], [0.5910, 0.4858]],
+                    ],
+                ]
+            ]
+        )
+        assert_expected(actual, expected, rtol=0, atol=1e-4)
+
+    def test_res_stack_has_attn(self, encoder):
```

Comment:
do we need separate tests for this and below?

Reply:
I'll delete `res_stack_has_attn` but I'd like to keep the other to test that the structure is correct